spec/features/authentication/set_up_enhanced_two_factor_authentication_spec.rb

```diff
@@ -0,0 +1,149 @@
+describe "Set up two factor authentication", type: :feature do
+  let(:organisation) { create(:organisation) }
+  let(:super_admin_organisation) { create(:organisation, super_admin: true) }
+  let(:user) { create(:user, organisations: [super_admin_organisation]) }
+
+  before do
+    allow(Rails.application.config).to receive(:enable_enhanced_2fa_experience).and_return true
+    sign_in_user(user, pass_through_two_factor: false)
+    visit root_path
+  end
+
+  it "presents the setup page" do
+    expect(page).to have_current_path("/users/two_factor_authentication/setup")
+  end
+
+  it "explains the setup step" do
+    expect(page).to have_content("Set up two-factor authentication")
+  end
+
+  it "provides the option to receive 2FA codes via email" do
+    expect(page).to have_css("#email")
+  end
+
+  it "provides the option to generate 2FA codes by authentication app" do
+    expect(page).to have_css("#app")
+  end
+
+  context "when navigating to another page" do
+    before { visit logs_path }
+
+    it "redirects the user back to setup" do
+      expect(page).to have_current_path("/users/two_factor_authentication/setup")
+    end
+  end
+
+  #########
+  # Email #
+  #########
```

Comment:
as much as I appreciate the art, can we leave it to the `context` block?

Reply:
😭